sources/subs/Cache.class.php

```diff
@@ -57,6 +57,12 @@ class Cache
 	protected $_accelerator = null;
 
 	/**
+	 * Cached keys
+	 * @var string[]
+	 */
+	protected $_cached_keys = array();
```

Comment:
The class is instantiated inside a singleton, so it should not run twice (unless some external code instantiate a new caching method by itself, in which case I'm not sure I would have this variable shared).

TBH, this is not the best solution either, I would have preferred if each caching had it's own way to deal with removing single keys, but it's not the case. From my readings it is reliably possible only for the file-based caching (because you can go through the file names), that is the less relevant, while the others have **some** way to remove **some** keys, but no way to find and remove keys based on pattern recognition. :disappointed: